app/models/specific_tube_rack_creation.rb

```diff
@@ -0,0 +1,436 @@
+  # @param [TubeRack] new_tube_rack The new tube rack object to which the barcode will be reassigned.
+  # @param [String] tube_rack_barcode The barcode of the tube rack.
+  #
+  # @raise [StandardError] if the barcode is already in use by another type of labware.
+  #
+  # @return [void]
+  def redirect_existing_barcode(existing_barcode_record, new_tube_rack, tube_rack_barcode)
+    existing_labware = existing_barcode_record.labware
+
+    if existing_labware.is_a?(TubeRack)
+      existing_barcode_record.labware = new_tube_rack
+    else
+      error_message =
+        "The tube rack barcode '#{tube_rack_barcode}' is already in use by " \
+          'another type of labware, cannot create tube rack.'
+      raise StandardError, error_message
+    end
+  end
+
+  # Adds metadata to a tube rack.
+  #
+  # This method creates a new PolyMetadatum record with the provided metadata key and tube rack barcode.
+  # The metadatable_type and metadatable_id are set to the class and ID of the tube rack, respectively.
+  # If the metadata record fails to save, an error is raised.
+  #
+  # @param [String] metadata_key The key for the metadata.
+  # @param [String] tube_rack_barcode The barcode of the tube rack to be used as the metadata value.
+  # @param [TubeRack] tube_rack The tube rack object to which the metadata will be added.
+  #
+  # @raise [StandardError] if the metadata record fails to save.
+  #
+  # @return [void]
+  def add_tube_rack_metadata(metadata_key, tube_rack_barcode, tube_rack)
+    pm =
+      PolyMetadatum.new(
+        key: metadata_key,
+        value: tube_rack_barcode,
+        metadatable_type: tube_rack.class,
+        metadatable_id: tube_rack.id
+      )
+    return if pm.save
+
+    raise StandardError, "New metadata for tube rack (key: #{metadata_key}, value: #{tube_rack_barcode}) did not save"
+  end
+
+  # Creates racked tubes based on the provided attributes and associates them with a new tube rack.
+  #
+  # This method iterates over the array of racked tube attributes provided in the rack_attributes
+  # hash.
+  # For each set of tube attributes, it calls the create_racked_tube method to create and associate
+  # the tube with the new tube rack.
+  #
+  # @param [Array<Hash>] racked_tubes An array of hashes, each containing attributes for a racked tube.
+  # @param [TubeRack] new_tube_rack The new tube rack object to which the tubes will be associated.
+  #
+  # @raise [StandardError] if any of the tube creation processes fail.
+  #
+  # @return [void]
+  def create_racked_tubes(racked_tubes, new_tube_rack)
+    # iterate through the rack attributes to create each tube
+    racked_tubes.each { |tube_attributes| create_racked_tube(tube_attributes, new_tube_rack) }
+  end
+
+  def create_racked_tube(tube_attributes, new_tube_rack)
+    tube = create_tube(tube_attributes)
+    link_tube_to_rack(tube, new_tube_rack, tube_attributes[:tube_position])
```

Comment:
Doesn't feel like `link_tube_to_rack` needs to be a separate method - could just paste contents in here.

Reply:
Was probably result of rubocop saying a method was too large/complex and me asking copilot to split it up into smaller chunks.